<!-- -->
<launch>
    <arg name="gating_thres" default="9.22" />
    <arg name="gate_probability" default="0.99" />
    <arg name="detection_probability" default="0.9" />
    <arg name="distance_thres" default="99" />
    <arg name="life_time_thres" default="8" />
    <arg name="static_velocity_thres" default="3.0" />
    <arg name="prevent_explosion_thres" default="1000" />
    <arg name="tracker_input_topic" default="/detection/lidar_objects" />
    <arg name="tracker_output_jskbb_topic" default="/detection/lidar_tracker/bounding_boxes" />
    <arg name="tracker_output_detected_topic" default="/detection/lidar_tracker/objects" />
    <arg name="use_sukf" default="false" />

    <arg name="tracking_frame" default="/world" />

    <!-- rosrun lidar_tracker euclidean_cluster _points_node:="" -->
    <node pkg="imm_ukf_pda_track" type="imm_ukf_pda" name="imm_ukf_pda" output="screen">
        <param name="gating_thres"            value="$(arg gating_thres)" />
        <param name="gate_probability"        value="$(arg gate_probability)" />
        <param name="detection_probability"   value="$(arg detection_probability)" />
        <param name="distance_thres"          value="$(arg distance_thres)" />
        <param name="life_time_thres"         value="$(arg life_time_thres)" />
        <param name="static_velocity_thres"   value="$(arg static_velocity_thres)" />
        <param name="prevent_explosion_thres" value="$(arg prevent_explosion_thres)" />
        <param name="tracking_frame"          value="$(arg tracking_frame)" />
        <param name="use_sukf"                value="$(arg use_sukf)" />

        <remap from="/detection/lidar_objects"         to="$(arg tracker_input_topic)" />
        <remap from="/detection/lidar_tracker/bounding_boxes" to="$(arg tracker_output_jskbb_topic)" />
        <remap from="/detection/lidar_tracker/objects"       to="$(arg tracker_output_detected_topic)" />
    </node>

    <node pkg="detected_objects_visualizer" type="visualize_detected_objects" name="ukf_track_visualization_01"
          output="screen" ns="/detection/lidar_tracker"/>
</launch>
